src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -80,7 +83,7 @@ private async Task<string> GetRegionAsync(ICoreLogger logger)
             catch (Exception e)
             {
                 logger.Info("[Region discovery] Call to local imds failed." + e.Message);
-                throw;
+                throw new MsalClientException(MsalError.RegionDiscoveryFailed, MsalErrorMessage.RegionDiscoveryFailed);
```

Comment:
seems like you are already throwing a MsalClientException on line 73. you can check to see if the exception caught here is a msalException and simply rethrow the exception instead of throwing with new here. Also, whenever you throw a new exception after catching one in a catch block, you will overwrite the stack trace.

Reply:
Thanks for pointing that out. Here I want to catch any exceptions that comes while making the local imds call so that it specifically tells that region discovery failed. Will fix it.

Reply:
Updated.